### PR TITLE
Install Xcode tools if needed

### LINKED ISSUE #52: BUILD FAILED (OS X 10.11.3 using python-build 20151028)
last line: **zipimport.ZipImportError: can't decompress data; zlib not available**

I'm working off of a fresh laptop, with more details in the log file provided below. Any suggested next actions? I did not see this error in previous issues.

[20160222_geraldarthur.laptop.log.zip](https://github.com/18F/laptop/files/141932/laptop.log.zip)

### LINKED ISSUE #153: Script fails on Xcode
I'm consistently getting a failure at the point of installing xcode and the xcode CLTs:

```
Installing Command Line Tools (macOS Sierra version 10.12) for Xcode
Error downloading Command Line Tools (macOS Sierra version 10.12) for Xcode: “Command Line Tools (macOS Sierra version 10.12) for Xcode” could not be verified.
Done.

Error downloading updates.
==> /usr/bin/sudo /bin/rm -f /tmp/.com.apple.dt.CommandLineTools.installondemand.in-progress
Password:
==> /usr/bin/sudo /usr/bin/xcode-select --switch /Library/Developer/CommandLineTools
xcode-select: error: invalid developer directory '/Library/Developer/CommandLineTools'
Failed during: /usr/bin/sudo /usr/bin/xcode-select --switch /Library/Developer/CommandLineTools
failed
```
I think it may be due to xcode 9 dropping today.

### OS X version you are using
12.6

### Did you recently upgrade from a previous OS X version?
No

### Are you running the script on a fresh OS X installation?
Yes

### Did you have any of the following tools installed before running the script?
- Xcode No
- Apple's command line tools No
- Anything the script installs (see "What it sets up" in the README) - if anything was installed prior to xcode, it's probably installed ok.

### LINKED ISSUE #157: Xcode command line tool instructions
For the non-devs using this, it would be helpful to know the Xcode command line tool needs to be installed for the Laptop script to work properly.

I believe `xcode-select --install` will work (thanks @micahtaylor!)

### PR DESCRIPTION
**Pull request by mgwalker**
closes #157
closes #153
closes #52

If the user has already installed Homebrew but somehow doesn't have the Xcode command line tools (CLT) (maybe after an OS upgrade? 🤷‍♂️), the script now attempts to install the CLT itself.  Uses the same logic as Homebrew.

Not sure how to test this, though.  I took out the conditional and it did download and install the CLT, but I also already had it installed so I don't really know what would have happened in a more realistic scenario.

### CODE REVIEW

File: mac [hunk 1]
```diff
@@ -141,6 +141,21 @@ else
...
+if ! xcode-select -p > /dev/null ; then
+  # From Homebrew's code: https://github.com/Homebrew/install/blob/878b5a18b89ff73f2f221392ecaabd03c1e69c3f/install#L297
+  fancy_echo "Installing Xcode Command Line Tools..."
+  # This temporary file prompts the 'softwareupdate' utility to list the Command Line Tools
+  clt_placeholder="/tmp/.com.apple.dt.CommandLineTools.installondemand.in-progress"
+  sudo /usr/bin/touch "$clt_placeholder"
+  clt_label=$(/usr/sbin/softwareupdate -l | grep -B 1 -E "Command Line (Developer|Tools)" | awk -F"*" '/^ +\\*/ {print $2}' | sed 's/^ *//' | tail -n1 | xargs)
+  sudo /usr/sbin/softwareupdate -i "$clt_label"
+  sudo /bin/rm -f  "$clt_placeholder"
+  sudo /usr/bin/xcode-select --reset
```
Comment (mgwalker) [inline mac]: The Homebrew version uses `--switch` to point to a particular install location `/Library/Developer/CommandLineTools`, but that fails sometimes for some reason.  Using `--reset` tells the system to use its standard search to find the CLT.  I think it works?

**Comment by monfresh** [comment]
To test this, I would recommend asking someone if they can get their hands on a spare laptop that can be erased and re-imaged with the target macOS. I have a spare personal laptop that I've been using over the years to test both the 18F laptop script and another one I maintain. I keep the current and previous macOS on USB sticks for easy wiping and reinstallation.

**Comment by mgwalker** [comment]
If only we could use virtual machines in anything like a sensible way...

**Comment by mgwalker** [comment]
Setup a macOS High Sierra VM on a personal machine and tested the following scenarios:

- [x] Homebrew already installed but Xcode CLT not installed (this should never happen but 🤷‍♂️)
  - script skips Homebrew install step and then installs CLT
- [x] Xcode installed before running the script
  - script installs Homebrew which installs CLT, skips script's CLT install step
- [x] Xcode and Homebrew installed before running the script but CLT not installed (should also not happen)
  - script skips Homebrew install step and then installs CLT
- [x] install on brand new machine without Xcode, Homebrew, CLT, etc.
  - script installs Homebrew which installs CLT, skips script's CLT install step

**Comment by afeld** [comment]
Just tested on a fresh-ish machine: Homebrew installs XCode Command Line Tools.

```
$ /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
...
==> The Xcode Command Line Tools will be installed.

Press RETURN to continue or any other key to abort
```

I think we are all set here, but let me know if I am missing something!

**Pull request closed without merging**